### PR TITLE
Add missing dependency crun to the podman spec

### PR DESCRIPTION
**Pull request by dcermak**
quadlet uses crun as the runtime, but crun is not required in any way by podman

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by vrothberg — COMMENTED** [review]
@rhatdan @lsm5 PTAL

**Comment by lsm5** [comment]
```
Requires: oci-runtime
Recommends: crun
Requires: (crun if fedora-release-identity-server)
```

We've had that setup to allow people to use runc and remove crun if they wish. But if quadlet can only use crun and not runc, then I can change it to a hard dep on crun. I'd also like to know how I should update runc dep then.

**Review by lsm5 — COMMENTED** [review]
While I agree the dependency should be updated, I'd prefer not to do it in `podman.spec.rpkg`. It's best handled in containers-common-extra fedora package. That reminds me, I'll need to remove some other dependencies from here.

**Comment by vrothberg** [comment]
We should probably also explore why Quadlet has a hard dep on crun.

@alexlarsson @ygalblum FYI

**Comment by dcermak** [comment]
> We should probably also explore why Quadlet has a hard dep on crun.

It's hardcoded here:
https://github.com/containers/podman/blob/bac20d1917622fa45130427c78231e2529ba328e/pkg/systemd/quadlet/quadlet.go#L317-L321

**Comment by lsm5** [comment]
> > While I agree the dependency should be updated, I'd prefer not to do it in podman.spec.rpkg. It's best handled in containers-common-extra fedora package
> 
> Any particular reason for that? Burying direct dependencies in dependent spec files can make it very easy to forget to update them if dependencies are removed or added.

There are many dependencies common to both buildah and podman, and also some to skopeo.

The podman.spec.rpkg is not the official fedora spec. It generates builds on [rhcontainerbot/podman-next](https://copr.fedorainfracloud.org/coprs/rhcontainerbot/podman-next/) . So, it's a lot cheaper to list common dependencies for multiple packages hosted in multiple repos in 1 single common dependency package which is containers-common.

**Comment by ygalblum** [comment]
@alexlarsson added this hard coded setting in `.container` file. I've just noticed that I did not set it for `.kube` files. So, should decide what's the right approach here:
1. Force `crun` on both types
2. Allow the user to set the value. If so, what should be the default value? Should Quadlet even set it?

**Comment by dcermak** [comment]
> > > While I agree the dependency should be updated, I'd prefer not to do it in podman.spec.rpkg. It's best handled in containers-common-extra fedora package
> > 
> > 
> > Any particular reason for that? Burying direct dependencies in dependent spec files can make it very easy to forget to update them if dependencies are removed or added.
> 
> There are many dependencies common to both buildah and podman, and also some to skopeo.

quadlet is a podman only feature, so the requirement on `crun` would be only for podman and not necessarily for buildah and skopeo.

**Comment by vrothberg** [comment]
> Allow the user to set the value. If so, what should be the default value? Should Quadlet even set it?

I think that's the right thing to do.  Quadlet enforcing crun would otherwise ignore the settings in containers.conf.

**Comment by vrothberg** [comment]
> > We should probably also explore why Quadlet has a hard dep on crun.
> 
> It's hardcoded here:
> 
> https://github.com/containers/podman/blob/bac20d1917622fa45130427c78231e2529ba328e/pkg/systemd/quadlet/quadlet.go#L317-L321

@giuseppe would that work with runc as well?

**Comment by lsm5** [comment]
> > > > While I agree the dependency should be updated, I'd prefer not to do it in podman.spec.rpkg. It's best handled in containers-common-extra fedora package
> > > 
> > > 
> > > Any particular reason for that? Burying direct dependencies in dependent spec files can make it very easy to forget to update them if dependencies are removed or added.
> > 
> > 
> > There are many dependencies common to both buildah and podman, and also some to skopeo.
> 
> quadlet is a podman only feature, so the requirement on `crun` would be only for podman and not necessarily for buildah and skopeo.

@vrothberg does buildah never need crun ?

**Comment by vrothberg** [comment]
> @vrothberg does buildah never need crun ?

Any runtime would do the job for Buildah.  There is no hard dep on crun.

**Comment by lsm5** [comment]
> Any runtime would do the job for Buildah. There is no hard dep on crun.

Ack. Going by the kinda parallel conversation on this PR, if quadlet can use runc soon, then we can keep the dependency list as-is inside containers-common-extra. Please correct me if I'm wrong.

@dcermak btw, in containers-common fedora package, the `containers-common-extra` is only meant for podman and buildah. Skopeo depends only on the reduced dependencies in the main `containers-common` package. Sorry for not being clear earlier.

**Comment by lsm5** [comment]
> > Any runtime would do the job for Buildah. There is no hard dep on crun.
> 
> Ack. Going by the kinda parallel conversation on this PR, if quadlet can use runc soon, then we can keep the dependency list as-is inside containers-common-extra. Please correct me if I'm wrong.

if that's gonna take longer than expected, I'm fine with merging this PR. HTH.

**Review by lsm5 — APPROVED** [review]
Given that quadlet is a crun-only feature for now, I'm cool with this. This can be reverted whenever we have runc support.

/lgtm
/hold

**Comment by giuseppe** [comment]
> @giuseppe would that work with runc as well?

yes that works with runc too

**Comment by vrothberg** [comment]
> > @giuseppe would that work with runc as well?
> 
> yes that works with runc too

Thanks!  In that case, I suggest to remove these lines from Quadlet.  Then, we don´t need any change to the spec.

**Comment by alexlarsson** [comment]
crun is a better match for what quadlet does imho, but yeah, really there isn't I think any particular reason to hardcode it.

I don't think we really should allow per-quadlet-file options for runtime though. It seems like this would be more of a global option for all quadlet services? Or maybe just use whatever the systemwide default is for all podman runs.

**Comment by alexlarsson** [comment]
@giuseppe And log=passthrough work with runc too?

**Comment by edsantiago** [comment]
Please add:
```diff
diff --git a/test/system/252-quadlet.bats b/test/system/252-quadlet.bats
index 03b6497d7..b0cdc4cc5 100644
--- a/test/system/252-quadlet.bats
+++ b/test/system/252-quadlet.bats
@@ -16,6 +16,11 @@ function start_time() {
 function setup() {
     skip_if_remote "quadlet tests are meaningless over remote"
 
+    runtime=$(podman_runtime)
+    if [[ "$runtime" != "crun" ]]; then
+        skip "runtime is $runtime; quadlet only works with crun"
+    fi
+
     test -x "$QUADLET" || die "Cannot run quadlet tests without executable \$QUADLET ($QUADLET)"
 
     start_time
```

**Comment by giuseppe** [comment]
> @giuseppe And log=passthrough work with runc too?

I've not tried but it should work there as well, the handling is in conmon

**Comment by TomSweeneyRedHat** [comment]
Once @edsantiago 's concern is addressed, LGTM

**Comment by ygalblum** [comment]
Following the comments from @alexlarsson and @giuseppe about the ability to use other runtimes, I think we should not merge this change and instead remove the hardcoded setting in Quadlet. I'll open a PR for if

**Comment by dcermak** [comment]
> Following the comments from @alexlarsson and @giuseppe about the ability to use other runtimes, I think we should not merge this change and instead remove the hardcoded setting in Quadlet. I'll open a PR for if

Agreed, sounds like the much better solution.

**Pull request closed without merging**
